### PR TITLE
Make let generation the default.

### PR DESCRIPTION
**Pull request by jdalton**
Now that Node 4 is the min bar let by default would be rad. For the case of the REPL I'm guessing it could be configured `false` for REPL use. I'm missing the option toggle for the REPL use.

### CODE REVIEW

**Comment by jdalton** [comment]
I'm missing the option toggle for the REPL use.

**Comment by jdalton** [comment]
While working on this PR I noticed some nested import tests failing
```js
switch (i) {
case 0:
  import { a as x } from "./abc";
  break;
case 1:
  import { b as x } from "./abc";
  break;
}
```
because:
```
SyntaxError: Identifier 'x' has already been declared
```
which makes sense because it's `let`. This is something to keep in mind as a toggle for nested import statements.

File: repl/index.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ function wrap(name, optionsArgIndex) {
   vm[name] = function (code) {
     var options = arguments[optionsArgIndex];
     var filename = options && options.filename;
-    var args = [compile(code, filename)];
+    var args = [compile(code, filename, { generateLetDeclarations: false })];
     for (var i = 1; i < arguments.length; ++i) {
```
Comment (jdalton) [inline repl/index.js]: Being able to set options for the REPL is probably a good thing regardless of the current nested import `let` issue.

**Comment by benjamn** [comment]
Re: `switch` case scope collisions, I'm tempted to wrap the `case:` bodies with braces, though I guess that would be an observable semantic change. Where should the `let` declarations go, if we wanted to make your example work?

**Comment by jdalton** [comment]
> Where should the let declarations go, if we wanted to make your example work?

If possible it could work by using `let` for top level import and `var` for nested.

**Comment by jdalton** [comment]
I'll hold off on this PR until a few other things are in place.

**Pull request closed without merging**